app/src/views/private/components/files-navigation.vue

```diff
@@ -0,0 +1,172 @@
+			<v-item-group v-model="openFolders" scope="files-navigation" multiple>
+				<v-list-group
+					clickable
+					:active="!currentFolder && !currentSpecial"
+					value="root"
+					scope="files-navigation"
+					exact
+					disable-groupable-parent
+					:arrow-placement="nestedFolders && nestedFolders.length > 0 ? 'after' : false"
+					@click="onClick({})"
+				>
+					<template #activator>
+						<v-list-item-icon>
+							<v-icon name="folder_special" outline />
+						</v-list-item-icon>
+						<v-list-item-content>
+							<v-text-overflow :text="t('file_library')" />
+						</v-list-item-content>
+					</template>
+
+					<navigation-folder
+						v-for="folder in nestedFolders"
+						:key="folder.id"
+						:click-handler="onClick"
+						:folder="folder"
+						:current-folder="currentFolder"
+						:actions-disabled="actionsDisabled"
+					/>
+				</v-list-group>
+			</v-item-group>
+		</div>
+
+		<v-divider />
+
+		<v-list-item clickable :active="currentSpecial === 'all'" @click="onClick({ special: 'all' })">
+			<v-list-item-icon><v-icon name="file_copy" outline /></v-list-item-icon>
+			<v-list-item-content>
+				<v-text-overflow :text="t('all_files')" />
+			</v-list-item-content>
+		</v-list-item>
+
+		<v-list-item clickable :active="currentSpecial === 'mine'" @click="onClick({ special: 'mine' })">
+			<v-list-item-icon><v-icon name="folder_shared" /></v-list-item-icon>
+			<v-list-item-content>
+				<v-text-overflow :text="t('my_files')" />
+			</v-list-item-content>
+		</v-list-item>
+
+		<v-list-item clickable :active="currentSpecial === 'recent'" @click="onClick({ special: 'recent' })">
+			<v-list-item-icon><v-icon name="history" /></v-list-item-icon>
+			<v-list-item-content>
+				<v-text-overflow :text="t('recent_files')" />
+			</v-list-item-content>
+		</v-list-item>
+	</v-list>
+</template>
+
+<script setup lang="ts">
+import { useI18n } from 'vue-i18n';
+import { onUnmounted, ref, watch } from 'vue';
+import { useFolders, Folder } from '@/composables/use-folders';
+import NavigationFolder from './files-navigation-folder.vue';
+import { isEqual } from 'lodash';
+import { useRouter } from 'vue-router';
+import { Special } from '@/types/folders';
+
+type Target = { special?: Special; folder?: string };
+
+const router = useRouter();
+
+const props = defineProps<{
+	currentFolder?: string;
+	currentSpecial?: Special;
+	customTargetHandler?: (target: Target) => void;
+	resetOpenFolders?: boolean;
+	actionsDisabled?: boolean;
+}>();
+
+const { t } = useI18n();
+
+const { nestedFolders, folders, loading, openFolders, resetOpenFolders } = useFolders();
+const prevOpenFolders = ref();
+
+watch([() => props.currentFolder, loading], setOpenFolders, { immediate: true });
+
+onUnmounted(() => {
+	if (props.resetOpenFolders) {
+		openFolders.value = prevOpenFolders.value;
+	}
+});
+
+function onClick(target: Target) {
+	if (props.customTargetHandler) {
+		props.customTargetHandler(target);
+	} else {
+		const path = ['files'];
```

Comment:
super tiny thing, but would `paths` be more apt here to slightly indicate it's an array?